Add user registration endpoint

MeghaMuralidharan opened the pull request:
POST /auth/register endpoint implemented.
User data is now successfully persisted in the PostgreSQL database.
Password hashing is correctly applied using sha256_crypt.
Changed the base

MeghaMuralidharan committed:
feat(auth): Implement user registration endpoint (Issue 5)

MeghaMuralidharan committed:
Update session.py

Database export issue

MeghaMuralidharan committed:
Delete app/schemas/users.py

Unnecessary file

MeghaMuralidharan committed:
Delete app/core/settings.py

Unwanted file

MeghaMuralidharan committed:
Updated Issue 5: <brief description of changes>

MeghaMuralidharan committed:
Your commit message describing the changes

the-chosen-reject commented on app/main.py:
```diff
@@ -1,7 +1,31 @@
+
+
+# Temporary function to drop and create all tables
+async def create_db_and_tables():
```
@MeghaMuralidharan 

- Do not use `(Base.metadata.drop_all` or `Base.metadata.create_all`
- It will drop and recreate your tables each time you run the app.
- Only use alembic commands manually for migrations.
- Remove this

the-chosen-reject commented on app/routes/auth.py:
```diff
@@ -0,0 +1,43 @@
+    new_user = await create_user(db, user_in)
+    
+    # 3. Return the user data, using UserRead to omit the password
+    return new_user
```
@MeghaMuralidharan 

Always return a JSON serializable object as result of a router function. 
Here `new_user` is an sqlalchemy table object

the-chosen-reject replied:
@MeghaMuralidharan ignore this.
`response_model=UserRead` this has been specified in router decorator that converts ORM object `new_user` to pydantic `UserRead` object. My bad that I didn't notice this

the-chosen-reject commented on app/routes/auth.py:
```diff
@@ -0,0 +1,43 @@
+    """
+    
+    # 1. Validate unique username using the service function
+    existing_user = await get_user_by_username(db, username=user_in.username)
```
Add try catch blocks in router level to handle exceptions.

MeghaMuralidharan committed:
feat: Complete Issue 5 - User registration endpoint

- Fixed auth route to properly return UserRead schema using from_orm()
- Removed manual table creation from main.py (using Alembic migrations instead)
- Removed unused session.py file (session management handled elsewhere)
- Cleaned up schema formatting

MeghaMuralidharan committed:
updated auth.py
